### ISSUE TITLE
Factorial of two negatives

### ISSUE THREAD

**Issue by hayd** [JuliaLang/julia#9612]
```jl
julia> factorial(-1)
ERROR: DomainError
 in factorial_lookup at combinatorics.jl:26
 in factorial at combinatorics.jl:35

julia> factorial(-1, -1)
0
```

Should this DomainError [if k<0](https://github.com/JuliaLang/julia/blob/bc9c1fa1745fe4b0d26cac3f0d04ec1ecd81d37c/base/combinatorics.jl#L64)?

**Comment by jlapeyre** [comment]
DomainError seems more correct.

**Comment by jiahao** [comment]
Closed by #9634 #9824
